CHEWBBACA/SchemaEvaluator/schema_evaluator.py

```diff
@@ -0,0 +1,703 @@
+
+    alleles_lengths.sort()
+
+    # number of alleles
+    nr_alleles = len(alleles_lengths)
+
+    # minimum and maximum values
+    max_length = max(alleles_lengths)
+    min_length = min(alleles_lengths)
+
+    # size range
+    size_range = f"{min_length}-{max_length}"
+
+    # Summary statistics
+    median_length = round(statistics.median(alleles_lengths))
+    mode_length = Counter(alleles_lengths).most_common()[0][0]
+
+    genes_info = [
+        gene,
+        nr_alleles,
+        size_range,
+        median_length,
+        allele_ids,
+        alleles_lengths,
+        mode_length,
+    ]
+
+    return genes_info
+
+
+def gene_seqs_info_boxplot(schema_dir):
+    """Determines boxplot statistics.
+
+    Args:
+        genes_list (list): a list with names/paths for FASTA
+        files.
+    Returns:
+        genes_info (dict): a dict with a subdict for each input
+        gene file. Each subdict contains information about each
+        gene such as, mode of the allele sizes, number of alleles
+        and summary statistics (min, max, median and mode).
+    """
+
+    schema_files = [
+        os.path.join(schema_dir, file)
+        for file in os.listdir(schema_dir)
+        if ".fasta" in file
+    ]
+
+    json_to_file = {"boxplot_data": []}
+
+    for g in schema_files:
+        seq_generator = SeqIO.parse(g, "fasta")
+        alleles_lengths = [len(allele) for allele in seq_generator]
+        alleles_lengths.sort()
+
+        # locus name
+        locus_name = os.path.split(g)[1]
+
+        # number of alleles
+        nr_alleles = len(alleles_lengths)
+
+        # minimum and maximum values
+        loci_max = max(alleles_lengths)
+        loci_min = min(alleles_lengths)
+
+        # standard deviation
+        if nr_alleles > 1:
+            locus_sd = statistics.stdev(alleles_lengths)
+        else:
+            locus_sd = 0.0
+
+        # median
+        median_length = round(statistics.median(alleles_lengths))
+
+        # mean
+        mean_length = round(sum(alleles_lengths) / nr_alleles)
+
+        # q1 and q3
+        if nr_alleles > 1:
+            half = int(nr_alleles // 2)
+            q1 = statistics.median(alleles_lengths[:half])
+            q3 = statistics.median(alleles_lengths[-half:])
+        else:
+            q1 = alleles_lengths[0]
+            q3 = alleles_lengths[0]
+
+        json_to_file["boxplot_data"].append({
+            "locus_name": locus_name,
+            "nr_alleles": nr_alleles,
+            "max": loci_max,
+            "min": loci_min,
+            "sd": locus_sd,
+            "median": median_length,
+            "mean": mean_length,
+            "q1": q1,
+            "q3": q3
+        })
+
+    # Sort data by locus_name
+    for k in json_to_file:
+        json_to_file[k] = sorted(json_to_file[k], key=itemgetter("locus_name"))
+
+
+    return json_to_file
+
+
+# Functions that obtain the data for panel E
+def create_cds_df(schema_dir, translation_table):
+    """
+
```

Comment:
It is best to add docstrings for all functions so that we do not have to worry about that later.